liquigraph-core/src/main/java/org/liquigraph/core/io/xml/ImplicitSchemaValidator.java

```diff
@@ -50,8 +54,124 @@ private static Schema implicitSchema(String name) {
         }
     }
 
+    static private class InputSourceToLSInput implements LSInput {
+
+        private InputSource source;
+
+        private InputSourceToLSInput(InputSource source) {
+            this.source = source;
+        }
+
+        @Override
+        public Reader getCharacterStream() {
+            return source.getCharacterStream();
+        }
+
+        @Override
+        public void setCharacterStream(Reader characterStream) {
+            source.setCharacterStream(characterStream);
+        }
+
+        @Override
+        public InputStream getByteStream() {
+            return source.getByteStream();
+        }
+
+        @Override
+        public void setByteStream(InputStream byteStream) {
+            source.setByteStream(byteStream);
+        }
+
+        @Override
+        public String getStringData() {
+            // TODO Auto-generated method stub
+            return null;
+        }
+
+        @Override
+        public void setStringData(String stringData) {
+            // TODO Auto-generated method stub
+
+        }
+
+        @Override
+        public String getSystemId() {
+            return source.getSystemId();
+        }
+
+        @Override
+        public void setSystemId(String systemId) {
+            source.setSystemId(systemId);
+        }
+
+        @Override
+        public String getPublicId() {
+            return source.getPublicId();
+        }
+
+        @Override
+        public void setPublicId(String publicId) {
+            source.setPublicId(publicId);
+        }
+
+        @Override
+        public String getBaseURI() {
+            // TODO Auto-generated method stub
+            return null;
+        }
+
+        @Override
+        public void setBaseURI(String baseURI) {
+            // TODO Auto-generated method stub
+
+        }
+
+        @Override
+        public String getEncoding() {
+            return source.getEncoding();
+        }
+
+        @Override
+        public void setEncoding(String encoding) {
+            source.setEncoding(encoding);
+        }
+
+        @Override
+        public boolean getCertifiedText() {
+            // TODO Auto-generated method stub
+            return false;
+        }
+
+        @Override
+        public void setCertifiedText(boolean certifiedText) {
+            // TODO Auto-generated method stub
+        }
+
+    }
+
     private Validator validator(SchemaErrorHandler customErrorHandler) {
         Validator validator = schema.newValidator();
+        ChainedEntityResolver resolverChain = new ChainedEntityResolver();
```

Comment:
I don't think you need the resolver at all when there are no explicitly declared schemas.
Liquigraph should always assume that no schema == latest local schema (i.e. embedded `schema/changelog.xsd`.

Reply:
If the resolver isn't there then it always resolves over http - I don't think that necessarily a problem, if someone needs to use without the internet they can just specify the schema (which is probably better/more explicit anyway), but it won't be using the local schema (I think?).
Happy to take out the implicit schema handling; it sounds like it doesn't always work, and users have an option to be more explicit if they need to work without the internet.

Reply:
Right, I see. Ideally, implicit schema should always resolve to the local embedded file.

Reply:
OK, I think the change allows it to resolve locally, @sey does that sound right?  What was the security manager issue you hit?